### PR TITLE
Add commands switch-(up/down)-or-else-workspace

### PR DESCRIPTION
**Pull request by gustavcedersjo**
Add commands to switch to the window above (or below), but fall back to switching to the workspace above (or below) if there was no window above (or below).

Keyboard shortcuts are added as well, but without default keybindings.

Issue #847

### CODE REVIEW

**Comment by Lythenas** [comment]
The code looks fine. I will try it out tomorrow if I get time.

Also I think the github action is broken...

File: schemas/org.gnome.shell.extensions.paperwm.gschema.xml [hunk 1]
```diff
@@ -238,6 +238,14 @@
         <default><![CDATA[[]]]></default>
         <summary>Switch to the below window (no monitor boundary)</summary>
     </key>
+    <key type="as" name="switch-up-or-else-workspace">
+        <default><![CDATA[['']]]></default>
```
Comment (Lythenas) [inline schemas/org.gnome.shell.extensions.paperwm.gschema.xml]: @jtaala should this be `<![CDATA[['']]]>`or `<![CDATA[[]]]>`? It looks like we used it inconsistently.

Comment (jtaala) [reply schemas/org.gnome.shell.extensions.paperwm.gschema.xml]: Good pickup @Lythenas - I don't think it matters here much(?), but for consistency and intent, it should be the latter for no keybind applied.

**Comment by jtaala** [comment]
Thanks @gustavcedersjo!  Hopefully will have time to look at this one tonight.

**Comment by jtaala** [comment]
Hey @gustavcedersjo - nice work.  I like it.

Is everyone happy with the description of these shortcuts?:

`Switch to the above window (no workspace boundary)`
`Switch to the below window (no workspace boundary)`

it definitely fits the decription for `Switch to the above window (no monitor boundary)` etc.  Any simpler way to describe these? (let me know if anyone has any bright ideas there).

Otherwise, am approving this and will merge tomorrow evening.

Jay.

**Review by jtaala — APPROVED** [review]
Looks good.  Great work @gustavcedersjo.

**Comment by Lythenas** [comment]
> `Switch to the above window (no workspace boundary)`
> `Switch to the below window (no workspace boundary)`

Maybe `Switch to window or workspace above/below`? Since we are switching to another workspace and not just switching a window if there is no window above/below.

**Comment by gustavcedersjo** [comment]
When I first made the patch, I called it something along the lines of your suggestion, @Lythenas, but I changed it when I saw the other settings. The `switch-gobal-*` could also be called something like `Switch to window or monitor above/below`.

**Comment by jtaala** [comment]
How's this?

image

**Comment by jtaala** [comment]
P.S.@gustavcedersjo,  we've trying a thing atm where we put up a donation link from contributors of PRs etc. (if they want) - see https://github.com/paperwm/PaperWM/discussions/723#discussioncomment-8290949.

Feel free to message me or comment here if you would like to have a donation link added. (can also leave a comment on https://confluence.jaytaala.com/display/AJT/ - I won't make any comment there re this public btw) .

**Pull request merged**